### ISSUE TITLE
[com.google.android.apps.maps] Google Maps no direction identifier on current location

### ISSUE THREAD

**Issue by pc00per** [microg/GmsCore#1690]
**Affected app**
Name: Google Maps, Google Earth.
Package id: `com.google.android.apps.maps`, `com.google.earth`

**Describe the bug**
Wondering if there would be a direction identifier, but I always have plain blue dot as current location.

**To Reproduce**
That's the default behaviour for me always.

**Expected behavior**
![v4-460px-Get-Current-Location-on-Google-Maps-Step-2-Version-6](https://user-images.githubusercontent.com/83843264/158090589-7bb3851d-fc44-4877-a9dc-47487108fdd5.jpg)

**Screenshots**
![Screenshot_20220314-070421_Earth](https://user-images.githubusercontent.com/83843264/158090126-5975e16e-84d9-413e-970e-a0a5f28772d4.png)

**System**
Android Version: 11
Custom ROM: CrDroid 7.14

**microG**
microG Core version: 0.2.24.214816
microG Self-Check results: All ticked
NLP Backends: Apple WiFi, DegaVu LS, GSM LS, WiFi LS, nominatim Address Lookup.

**Comment by xk0niR** [comment]
I´ve got the same exact problem.

If you are using root:
1. Force-stop Maps and delete /data/data/com.google.android.apps.maps/files/SavedClientParameters.data.cs
2. Make /data/data/com.google.android.apps.maps/files read-only (551)

2. is necessary otherwise SavedClientParameters.data.cs will be created again and cause this error again.

**Comment by Jakeler** [comment]
Have the same problem. 
The trick from @xk0niR works, thanks! Only problem is that now it always uses the white theme and the theme selector does not pop up anymore. Allowing writes (771) to files immediately shows the dark mode mode again... Any way around that?

**Comment by pc00per** [comment]
@ale5000-git Thank you mate. Appreciate your generosity.

**Comment by connor-brooks** [comment]
I seem to be having a similar issue too, however I have no root...

**Comment by connor-brooks** [comment]
I am having the same issue too, but I have no root.

**Comment by xk0niR** [comment]
> I am having the same issue too, but I have no root.

I´m using Maps 10.87.4 from apkmirror (Architecture universal, nodpi) and it seems to work.
It´s complaining about missing Google Play Services but otherwise it´s working without root.

**Comment by Jakeler** [comment]
I noticed it also says `Location accuracy is low` and recommends to calibrate it with the camera. This works if you are in a city at daytime and then at least for some time the compass is working.

**Comment by connor-brooks** [comment]
> > I am having the same issue too, but I have no root.
> 
> I´m using Maps 10.87.4 from apkmirror (Architecture universal, nodpi) and it seems to work. It´s complaining about missing Google Play Services but otherwise it´s working without root.

I just tried this version and it still does the same thing. Strange.

**Comment by surfbro** [comment]
> I´ve got the same exact problem.
> 
> If you are using root:
> 
> 1. Force-stop Maps and delete /data/data/com.google.android.apps.maps/files/SavedClientParameters.data.cs
> 2. Make /data/data/com.google.android.apps.maps/files read-only (551)
> 
> It is necessary to perfom step 2. otherwise SavedClientParameters.data.cs will be created again and cause this error again.

As of today using this trick, it works temporarily. After force-stopping Maps, deleting SavedClientParameters.data.cs and then marking the /files directory as read-only using Octal 551 as r-xr-x--x , Maps functions with the gyroscope and turning normally. Giving back the walking directions that were missing.

However, after killing Maps and restarting, I go back into my File Manager and see that the directory /files returns to 771. And then within another kill/open of Maps, SavedClientParameters.data.cs has been created again. This is under the Maps version 10.87.4

Has anyone else figured out _another_ workaround?

**Comment by surfbro** [comment]
> > > I am having the same issue too, but I have no root.
> > 
> > 
> > I´m using Maps 10.87.4 from apkmirror (Architecture universal, nodpi) and it seems to work. It´s complaining about missing Google Play Services but otherwise it´s working without root.
> 
> I just tried this version and it still does the same thing. Strange.

When you say that it does the same thing, are you talking about missing Google Play Services? That is easily defeated by holding the notification, and then entering the expanded notifications allow/disallow under LineageOS. You can toggle off Other notifications by expanding the Notifcations under App Settings, and then you want to disable 'Google Play services availability'.

However, the issue still remains that the gyroscope functionality is disabled after the first boot of the app. Unless I am doing something wrong...

**Comment by JRuiBarros** [comment]
I'm facing the same issue as well, using the official APK. Like @surfbro mentioned, gyroscope works on first boot and then it stops working afterwards. I don't want to root my phone so a proper fix would be appreciated. I can provide more info if necessary.

**Comment by pc00per** [comment]
> I'm facing the same issue as well, using the official APK. Like @surfbro mentioned, gyroscope works on first boot and then it stops working afterwards. I don't want to root my phone so a proper fix would be appreciated. I can provide more info if necessary.

I think the issue is google's maps applications themselves. You can test gyroscope on Osmand & other map providers they work perfectly there.

**Comment by surfbro** [comment]
> > I'm facing the same issue as well, using the official APK. Like @surfbro mentioned, gyroscope works on first boot and then it stops working afterwards. I don't want to root my phone so a proper fix would be appreciated. I can provide more info if necessary.
> 
> I think the issue is google's maps applications themselves. You can test gyroscope on Osmand & other map providers they work perfectly there.

Yeah it 100% has something to do with Google punishing you for using alternative services. However, I wanted to confirm if anyone here has gyroscope working using the method above. From what I can tell other versions of Google Maps will rebuild the /files directory, rendering the workaround useless.

I noticed after one full day, that the /files directory was recreated by Maps, and then back to the same issue...

**Comment by pc00per** [comment]
@surfbro I think you need to charge `owner` & the `group` permissions of files folder to `root` It basically should look like this...
![Screenshot_20220724](https://user-images.githubusercontent.com/83843264/180631475-e62124f5-2f2c-437f-8f10-c4bc36678cf1.png)

**Comment by surfbro** [comment]
> @surfbro I think you need to change `owner` & the `group` permissions of files folder to `root` It basically should look like this...
> 
> ![Screenshot_20220724](https://user-images.githubusercontent.com/83843264/180631475-e62124f5-2f2c-437f-8f10-c4bc36678cf1.png)

Thanks! I set the permissions and will report back in a day to confirm the Gmaps accelerometer/gyroscope still works. I noticed my previous method would work under 24 hr, and then would reset itself. So Ill give it another 24 hr and report back!

**Comment by pc00per** [comment]
@surfbro Also, a shorter & better way would be to just remove read permission for `SavedClientParameters.data.cs`

![Screenshot_20220727](https://user-images.githubusercontent.com/83843264/181162651-7f76289d-76bc-421e-a56d-b291eaf5d369.png)

**Comment by surfbro** [comment]
> @surfbro I think you need to change `owner` & the `group` permissions of files folder to `root` It basically should look like this...
> 
> ![Screenshot_20220724](https://user-images.githubusercontent.com/83843264/180631475-e62124f5-2f2c-437f-8f10-c4bc36678cf1.png)

THIS WORKED! Thanks for coming back here to recommend these ownership settings as well.

**Comment by Rmtony** [comment]
I am on CalyxOs 3.9.0 Android 12L , no root, and I have this issue.... any fix \ solution ?

**Comment by pc00per** [comment]
> I am on CalyxOs 3.9.0 Android 12L , no root, and I have this issue.... any fix \ solution ?

No unfortunately. You're using a custom rom, so I better suggest to root. An unrooted phone is like a linux admin without root previleges for me.

**Comment by Rmtony** [comment]
> > I am on CalyxOs 3.9.0 Android 12L , no root, and I have this issue.... any fix \ solution ?
> 
> No unfortunately. You're using a custom rom, so I better suggest to root. An unrooted phone is like a linux admin without root previleges for me.

for security reasons it is better to don't root

**Comment by Rmtony** [comment]
Today I flashed another phone, Pixel 3a, and I confirm the issue... even the latest microG release did not solve the problem...

**Comment by pc00per** [comment]
> for security reasons it is better to don't root

Rooting is the most secure process * if you know what you're doing. It actually more secure than being non root. Because, you being an admin of your device don't have privileges to do modify your own device, that's like you running a Linux machine without having sudo privileges.
> even the latest microG release did not solve the problem

It's not a microg issue. It's the google apps themselves.

**Comment by Jakeler** [comment]
> > even the latest microG release did not solve the problem
> 
> It's not a microg issue. It's the google apps themselves.

It is absolutely a microG issue - Google Maps works normally with the proprietary Play Services. So the app would not detect it if microG would implement all APIs 100% perfectly. Even if you like rooting, disabling writes on app data is just a dirty hack with unforeseeable side effects (one already known is the dark mode), this can also completely break the app on any update.
That being said, this is a relatively minor issue, affects only GMaps compass, otherwise it the app works, also OsmAnd, HERE WeGo etc. work normally with compass data. Implementing google APIs is a giant project, will be practically impossible to get to 100%. I am incredibly thankful for @mar-v-in and all other contributors putting so much work into it. It is usable as daily driver and frees us from hundreds of MBs of proprietary software.

**Comment by pc00per** [comment]
@Jakeler You can test the compass on osmand, openstreetmap & maps.me. Works on all of em for me.

**Comment by Rmtony** [comment]
> > for security reasons it is better to don't root
> 
> Rooting is the most secure process * if you know what you're doing. It actually more secure than being non root. Because, you being an admin of your device don't have privileges to do modify your own device, that's like you running a Linux machine without having sudo privileges.

Ask to CalyxOS or Graphene teams if it secure or not....

> 
> > even the latest microG release did not solve the problem
> 
> It's not a microg issue. It's the google apps themselves.

It is a microG issue! That said MicroG works great, compass in Gmaps is the only issue at least for me.
And lastly. are you with us ? On planet earth? :)

**"MicroG allows Android apps to access replica application programming interfaces (APIs) that are provided by Google Play Services, including the APIs associated with Google Play, Google Maps, and Google's geolocation and messaging features. Unlike Google Play Services, MicroG does not track user activity on the device, and users can selectively enable and disable specific API features. Depending what apps are installed by users, user activity may still be tracked by Google."**

**Comment by pc00per** [comment]
> Ask to CalyxOS or Graphene teams if it secure or not....

If you haven't tried magisk, it can actually allow or deny root access to any application on the fly. CalyxOS/Graphene are just a fork of AOSP, where both are vanilla only variants. There happens to have zillions of AOSP roms that claim to be secure, but the true fact is any AOSP rom that's open source, is always secure.
> It is a microG issue! That said MicroG works great, compass in Gmaps is the only issue at least for me.
And lastly. are you with us ? On planet earth? :)

It's not an issue, since other open source map clients able to utilize smartphone's resources (compass in this case) without hickups.
Again, Google applications are not open source, so there will never be any workaround to this.

**Comment by Rmtony** [comment]
> > Ask to CalyxOS or Graphene teams if it secure or not....
> 
> If you haven't tried magisk, it can actually allow or deny root access to any application on the fly. CalyxOS/Graphene are just a fork of AOSP, where both are vanilla only variants. There happens to have zillions of AOSP roms that claim to be secure, but the true fact is any AOSP rom that's open source, is always secure. Those two happened to have some marketing buzz, so you happened to notice only those two.
> 
> > It is a microG issue! That said MicroG works great, compass in Gmaps is the only issue at least for me.
> > And lastly. are you with us ? On planet earth? :)
> 
> It's not an issue, since other open source map clients able to utilize smartphone's resources (compass in this case) without hickups. Again, Google applications are not open source, so there will never be any workaround to this.

Other maps works because are open source, you answered yourself, I repeat, it is a MicroG issue, you don't even know what MicrogG is. Then if it is a Google issue(they Must implement microG instead of their own API .. Hahahahhah :)  ) why you opened an issue in MicroG section of Github ? Are you confused ?

**Comment by pc00per** [comment]
@Rmtony That's true. But these issues have to be brought somewhere so that they're recognized. There's already a root workaround being suggested by one above due to this. 
Google can't do a thing about it. Caz it only happens to occur on microg. Their play services & their application set both are closed. If you wanna expect pitch perfect workflow, it's better to stay on play services.

**Comment by ale5000-git** [comment]
This issue can be solved in microG (or in the mapbox library it use, I don't know exactly).
The only problem is that there is just one active developer so it may take 1 day or even 15 years, who know.

**Comment by pc00per** [comment]
Gmaps seems to store client data & execute `SavedClientParameters.data.css` when application opened. But fails to initialize direction radar. That's probably not the case when the file doesn't exist, which is the reason radar works again when it gets removed. It's the gmaps side behaviour afaik, as it's not able to get the client data after it's been stored.

**Comment by ale5000-git** [comment]
A lot of methods inside microG are stubs, so it may be one of them is needed to be working otherwise GMaps get confused anf fail to initialize (just a supposition).

**Comment by Rmtony** [comment]
Someone tried latest MicroG release v0.2.26.223616 ?

**Comment by mar-v-in** [comment]
This is still a pending issue in 0.2.26.

**Comment by Rmtony** [comment]
what about v0.2.27.223616 ? fixed ?

**Comment by LuccoJ** [comment]
I don't know why some of you had to make such drama. "Other" open source applications, for example, may not be using and expecting the Google Fused Location provider, which I think microG emulates but maybe not completely. Calm down a little bit and don't ask people what planet they're from as that is not at all relevant, please.
